src/main.c

```diff
@@ -42,7 +48,31 @@ int main(int argc, char **argv)
 	while (true) {
 		volatile exception_s e;
 		TRY_CATCH (e, EXCEPTION_ALL) {
-			gdb_main();
+			SET_IDLE_STATE(false);
+			while (gdb_target_running && cur_target) {
```

Comment:
Is it possible to flatten this out a bit by making this while loop the outer while loop?